server/task/task-memory/src/main/scala/org/apache/james/task/eventsourcing/EventSourcingTaskManager.scala

```diff
@@ -90,7 +90,6 @@ class EventSourcingTaskManager @Inject @VisibleForTesting private[eventsourcing]
 
   private def listScala: List[TaskExecutionDetails] = executionDetailsProjection
     .list
-    .flatMap(details => executionDetailsProjection.load(details.taskId))
```

Comment:
where the projection is computed?
Also, `getExecutionDetails` should not call `load`neither?

Reply:
`executionDetailsProjection.load(details.taskId))` was calling directly the projection get(taskId)

Hence, given the meaningfulness of this load method I don't see why we should keep that unoptimized code path.

Reply:
I didn't understand your answer.
I read the code again and I now can answer my own questions:

> where the projection is computed?

It's actually stored into Cassandra (or in-memory) on new events

> Also, getExecutionDetails should not call `load` neither?

`list` does exactly the same thing than `load` but for the whole list of tasks.